src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathCheckSpec.groovy

```diff
@@ -14,44 +14,53 @@ import static org.openkilda.model.FlowEncapsulationType.VXLAN
     @Tidy
     @Tags(SMOKE)
     def "No path validation errors for valid path without limitations"() {
         given: "Path for non-neighbouring switches"
-        def path = topologyHelper.getNotNeighboringSwitchPair().getPaths().sort {it.size()}.first()
+        def path = topologyHelper.getAllSwitchPairs().nonNeighbouring().random()
+                .getPaths().sort { it.size() }.first()
 
         when: "Check the path without limitations"
-        def validationErrors = pathHelper.'get path check errors'(path)
+        def pathCheckResult = pathHelper.getPathCheckResult(path)
 
         then: "Path check result doesn't have errors"
-        validationErrors.isEmpty()
+        verifyAll(pathCheckResult) {
+            getErrors().isEmpty()
+            getPceResponse() == PCE_PATH_COMPUTATION_SUCCESS_MESSAGE
+        }
+
     }
 
     @Tidy
     @Tags(SMOKE)
     def "Path check errors returned for each segment and each type of problem"() {
         given: "Path of at least three switches"
-        def switchPair = topologyHelper.getAllNotNeighboringSwitchPairs().shuffled().first()
+        def switchPair = topologyHelper.getAllSwitchPairs().nonNeighbouring().random()
         def path = switchPair.getPaths()
-                .sort {it.size()}
+                .sort { it.size() }
                 .first()
         def srcSwitch = switchPair.getSrc()
 
         and: "Source switch supports only Transit VLAN encapsulation"
         def backupSwitchProperties = switchHelper.getCachedSwProps(srcSwitch.getDpId())
         switchHelper.updateSwitchProperties(srcSwitch, backupSwitchProperties.jacksonCopy().tap {
-                it.supportedTransitEncapsulation = [TRANSIT_VLAN.toString()]
-            })
+            it.supportedTransitEncapsulation = [TRANSIT_VLAN.toString()]
+        })
 
         when: "Check the path where one switch doesn't support flow encapsulation type and all links don't have enough BW"
-        def validationErrors = pathHelper.'get path check errors'(path, Long.MAX_VALUE,  VXLAN)
+        def pathCheckResult = pathHelper.getPathCheckResult(path, Long.MAX_VALUE, VXLAN)
 
         then: "Path check result has multiple lack of bandwidth errors and at least one encapsulation support one"
-        !validationErrors.findAll{it == "The switch ${srcSwitch.getDpId()} doesn\'t support the encapsulation type VXLAN."}.isEmpty()
-        validationErrors.findAll {it.contains("not enough bandwidth")}.size() ==
-                pathHelper.getInvolvedIsls(path).size() * 2
+        verifyAll(pathCheckResult) {
+            !getErrors().findAll { it == "The switch ${srcSwitch.getDpId()} doesn\'t support the encapsulation type VXLAN." }.isEmpty()
+            getErrors().findAll { it.contains("not enough bandwidth") }.size() ==
+                    pathHelper.getInvolvedIsls(path).size() * 2
+            getPceResponse().contains("Failed to find path with requested bandwidth")
```

Comment:
WDYT to add an enum for the PCE_PATH_COMPUTATION responses (PCEPathComputationReport) for successful response and limitation for bandwidth or latency?

Reply:
Doesn't worth it, IMHO. This response is checked only in these tests and it's unlikely to be used/verified in another tests.